test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -6252,7 +6253,35 @@ def submit(self, obj, script=None, extend=None, submit_dir=None):
         rc = None
 
         if isinstance(obj, Job):
-            if script is None and obj.script is not None:
+            if self.platform == 'cray' or self.platform == 'craysim':
+                m = None
+                vncompute = False
+                if 'Resource_List.select' in obj.attributes:
+                    m = re.match('.*vntype=cray_compute.*',
+                                 obj.attributes['Resource_List.select'])
+                if 'Resource_List.vntype' in obj.attributes:
+                    vn_type = obj.attributes['Resource_List.vntype']
+                    if vn_type == 'cray_compute':
+                        vncompute = True
+                if obj.script is not None:
+                    script = obj.script
+                elif m is not None or vncompute:
+                    aprun_cmd = "aprun -B"
+                    if script:
+                        aprun_cmd += " " + script
```

Comment:
Here script is nothing but the path to application(mpp compiled for cray, example: hello.sleep) which needs to be run with aprun command and this path user can specify through script parameter of submit function.